src/Provider/AbstractProvider.php

```diff
@@ -387,32 +387,62 @@ protected function verifyGrant($grant)
     }
 
     /**
-     * Returns a prepared request for requesting an access token.
+     * Returns the full URL to use when requesting an access token.
      *
-     * @param array $params Query string parameters
+     * @param array $params Query parameters
      */
-    protected function getAccessTokenRequest(array $params)
+    protected function getAccessTokenUrl(array $params)
     {
         $url = $this->getBaseAccessTokenUrl();
-        $query = $this->getAccessTokenQuery($params);
-        $method = strtoupper($this->getAccessTokenMethod());
 
+        if ($this->getAccessTokenMethod() === self::METHOD_GET) {
+            $query = $this->getAccessTokenQuery($params);
+            return $this->appendQuery($url, $query);
+        }
+
+        return $url;
+    }
+
+    /**
+     * Returns the request body for requesting an access token.
+     *
+     * @param  array $params
+     * @return string
+     */
+    protected function getAccessTokenBody(array $params)
```

Comment:
This is *almost* identical to `getAccessTokenQuery`. I would be okay with combining them, but I don't want to use `$options['body'] = $query`.